plutus-core/plutus-ir/test/transform/knownCon/maybe-unsaturated

```diff
@@ -0,0 +1,25 @@
+(let
+  (nonrec)
+  (datatypebind
+    (datatype
+      (tyvardecl Maybe (fun (type) (type)))
+      (tyvardecl a (type))
+      Maybe_match
+      (vardecl Just (fun a [ Maybe a ])) (vardecl Nothing [ Maybe a ])
+    )
+  )
+  [
+    {
+      [
+        { Maybe_match (con integer) } [ { Just (con integer) } (con integer 1) ]
+      ]
+      (all dead (type) (con integer))
+    }
+    (lam
+      a
+      (con integer)
+      (abs dead (type) [ [ (builtin addInteger) (con integer 2) ] a ])
+    )
+  ]
+)
+-- `Maybe_match` is unsaturated. This should never happen in practice.
```

Comment:
put it at the top?

also I want to see an over-saturated case! where the match returns a function

```
(case (Just addInteger) of 
   Just f -> f
   Nothing -> subtractInteger) 1 2
```

Reply:
Most if not all test cases are over-saturated, the additional argument being `(all dead (type) dead)`.

Reply:
Aha, good point! Worth calling out explicitly.